compiler/testData/codegen/box/functions/bigArity/callFunViaVararg.kt

```diff
@@ -1,6 +1,7 @@
 // !LANGUAGE: +FunctionTypesWithBigArity
```

Comment:
What happens to this test?
Do we not want to support this kind of code any more?
If so, the test should probably just be removed.

Reply:
That was my understanding based on [KT-32701](https://youtrack.jetbrains.com/issue/KT-32701). Instead of removing the test, I've changed it to have only a single large arity supertype.

Going forward we still want to support this case, at least after [KT-32698](https://youtrack.jetbrains.com/issue/KT-32698) is fixed.